controllers/rabbitmq/transporturl_controller.go

```diff
@@ -0,0 +1,267 @@
+			err.Error()))
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	password, ctrlResult, err := secret.GetDataFromSecret(ctx, helper, rabbit.Status.DefaultUser.SecretReference.Name, 10, "password")
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			rabbitmqv1beta1.TransportURLReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	host, ctrlResult, err := secret.GetDataFromSecret(ctx, helper, rabbit.Status.DefaultUser.SecretReference.Name, 10, "host")
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			rabbitmqv1beta1.TransportURLReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// Create a new secret with the transport URL for this CR
+	secret := r.createTransportURLSecret(instance, string(username), string(password), string(host))
+	_, op, err := oko_secret.CreateOrPatchSecret(ctx, helper, instance, secret)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			rabbitmqv1beta1.TransportURLReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	if op != controllerutil.OperationResultNone {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			rabbitmqv1beta1.TransportURLReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			rabbitmqv1beta1.TransportURLReadyInitMessage))
+		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
+	}
+
+	// Update the CR and return
+	instance.Status.SecretName = secret.Name
+
+	instance.Status.Conditions.MarkTrue(rabbitmqv1beta1.TransportURLReadyCondition, rabbitmqv1beta1.TransportURLReadyMessage)
+
+	return ctrl.Result{}, nil
+
+}
+
+// Create k8s secret with transport URL
+func (r *TransportURLReconciler) createTransportURLSecret(instance *rabbitmqv1beta1.TransportURL, username string, password string, host string) *corev1.Secret {
+	// Create a new secret with the transport URL for this CR
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rabbitmq-transport-url-" + instance.Name,
+			Namespace: instance.Namespace,
+		},
+		Data: map[string][]byte{
+			"transport_url": []byte(fmt.Sprintf("rabbit://%s:%s@%s:5672", username, password, host)),
+		},
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *TransportURLReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&rabbitmqv1beta1.TransportURL{}).
```

Comment:
I think we need `Owns(&Sercret{})` here too as this controller creates Secret objects.

Reply:
good call its in the latest commit